tortoisebeacon/weakcoin/weak_coin.go

```diff
@@ -131,41 +152,49 @@ func (wc *weakCoin) PublishProposal(ctx context.Context, epoch types.EpochID, ro
 		return fmt.Errorf("broadcast weak coin message: %w", err)
 	}
 
+	wc.Log.With().Info("Published proposal",
+		log.Uint64("epoch_id", uint64(epoch)),
+		log.Uint64("round_id", uint64(round)),
+		log.String("proposal", types.BytesToHash(signedProposal).ShortString()))
+
 	return nil
 }
 
 func (wc *weakCoin) proposalExceedsThreshold(proposal []byte) bool {
-	proposalInt := new(big.Int).SetBytes(proposal[:])
+	proposalInt := new(big.Int).SetBytes(proposal)
```

Comment:
bytes.Compare ?

Reply:
As `SetBytes` expects a big-endian number, it looks to me like another way to do the same. Feel free to change if you want